mne/io/edf/edf.py

```diff
@@ -572,7 +571,28 @@ def _read_edf_header(fname, exclude):
                 edf_info['units'].append(1e-6)
             else:
                 edf_info['units'].append(1)
+
         ch_names = [ch_names[idx] for idx in sel]
+        units = [units[idx] for idx in sel]
+        # make sure channel names are unique
+        # code modified from _check_consistency function in meas_info.py
+        unique_ids = np.unique(ch_names, return_index=True)[1]
+        if len(unique_ids) != len(ch_names):
+            dups = set(ch_names[x]
+                       for x in np.setdiff1d(range(len(ch_names)), unique_ids))
+            warn('Channel names are not unique, found duplicates for: '
+                 '%s. Applying running numbers for duplicates.' % dups)
+            for ch_stem in dups:
+                overlaps = np.where(np.array(ch_names) == ch_stem)[0]
+                n_keep = min(len(ch_stem),
+                             14 - int(np.ceil(np.log10(len(overlaps)))))
```

Comment:
Maybe add a comment why this magic name truncation was chosen (why 14 and why based on the log of the overlaps)?